### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by PotatoCombat**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -134,24 +135,155 @@ The `UI` component,
...
+
+========
+In future versions, `Prefixes` will be stored in the *Commands* package,
+since each command will need to know all the `Prefix` that they are using.
+
+This means that:
+
+* The command parsers will obtain all the required `Prefix` from the respective command.
+* The *green arrow* can be removed indicating that the command parsers no longer need information in `Prefixes`.
+========
```
Comment (chuayijing) [inline docs/DeveloperGuide.adoc]: mayb can write it as design consideration??? looks better i think kekeke

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -203,14 +336,25 @@ The `Storage` component,
...
-** each `Module` contains a list of `Session`, representing the sessions that
-the user has completed for that module.
-** each `Module` contains a list of `Group`, representing the groups that the user is
-in charge of, such as a tutorial or lab.
-** each `Group` contains a list of `Student`, representing the students enrolled in
+* a list of `Module` objects representing the modules that the user is assisting. +
++
+Within each module, there is:
+
+** a list of `Session`, representing the `done sessions` that the user has completed for that module.
```
Comment (chuayijing) [inline docs/DeveloperGuide.adoc:380]: does it not save the un-done sessions under Session view?

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -575,119 +719,172 @@ image::TssActivityDiagram.png[]
...
-| Valid inputs may be obscured due to the choice of colour.
+Here is the purpose of each partition in the above diagram:
+
+* `wrong` +
+When the command does not need the given `argument`,
+the `ResultBox` should reshow how to use the command.
+
+* `invalid` +
+When the `argument` cannot have the given `value`,
+the `ResultBox` should show how to use the `argument`.
```
Comment (chuayijing) [inline docs/DeveloperGuide.adoc]: mayb can give example at the side for more clarity

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (PotatoCombat) [reply docs/DeveloperGuide.adoc:380]: It does! There are two lists! One for modules, and one for undone sessions

**Review by chuayijing — APPROVED** [review]

**Pull request merged**